go-controller/pkg/factory/factory.go

```diff
@@ -666,7 +666,7 @@ func NewNodeWatchFactory(ovnClientset *util.OVNNodeClientset, nodeName string) (
 	}
 
 	// need to configure OVS interfaces for Pods on secondary networks in the DPU mode
-	if config.OVNKubernetesFeature.EnableMultiNetwork && config.OvnKubeNode.Mode == types.NodeModeDPU {
+	if config.OVNKubernetesFeature.EnableMultiNetwork {
```

Comment:
so for secondary networks we don't need this cause we don't watch for NADs in node watch factory huh?

So its really:
(config.OVNKubernetesFeature.EnableMultiNetwork && config.OvnKubeNode.Mode == types.NodeModeDPU) || util.IsNetSegEnabled() && !Interconnect?

Reply:
yeah I guess. 

Not easy to reason about that or explain why though ...

Reply:
nit: adding a comment to that effect would be good, but I can trace it back